### PR TITLE
feat: allow user to run a dryrun with and without a signer

### PR DESCRIPTION
**Pull request by rvermootenct**
---
name: Allow user to run dry run with and without a signer
about: A user girl should be able to decide if she wants to run dry mode with or without a signer present. 
labels: 

---
IMPORTANT NOTICE:
I read and understood the [guidelines for contributions to the TRD](https://tezos-reward-distributor-organization.github.io/tezos-reward-distributor/contributors.html). The contribution may qualify for being compensated by the TRD grant if approved by the maintainers.

This PR resolves the issue [614](https://github.com/tezos-reward-distributor-organization/tezos-reward-distributor/issues/614). The following steps were performed:

* **Analysis**:  It was determined that there are cases where it is both helpful and annoying to have to use a signer during dry run mode. 

* **Solution**:  Allow a user girl to decide whether she would like to use a signer or not depending on their use case. 

* **Implementation**: Move what had been done in PR [615](https://github.com/tezos-reward-distributor-organization/tezos-reward-distributor/pull/615) into its own option, allowing dry run to stay as it was before. 

* **Performed tests**:  Ran dry run and dry run no signer locally. 

* **Documentation**:  Updated the help documentation, 

* **Check list**:

- [ ] I extended the Github Actions CI test units with the corresponding tests for this new feature (if needed).
- [ ] I extended the Sphinx documentation (if needed).
- [x] I extended the help section (if needed).
- [ ] I changed the README file (if needed).
- [ ] I created a new issue if there is further work left to be done (if needed).

**Work effort**: 1

* **GIF tax**:

### CODE REVIEW

File: docs/run.rst [hunk 1]
```diff
@@ -59,6 +59,9 @@ Options
 ``-D --dry_run``
     Run without injecting payments. Suitable for testing. Does not require locking.
```
Comment (jdsika) [inline docs/run.rst:60]: Idea: Move option "Dp" up to group all "dry-run-options". Create a small glossary like p=no publish stats and s=no signer and make clear that there are possibilities how those options can be combined.

File: src/launch_common.py [hunk 2]
```diff
@@ -147,6 +147,12 @@ def args_validation(args, argparser):
         else:
             args.dry_run = args.dry_run_no_consumers
     else:
+        try:
```
Comment (jdsika) [inline src/launch_common.py:150]: How those options impact each other and how they can be combined confuses me slightly

File: src/launch_common.py [hunk 3]
```diff
@@ -147,6 +147,12 @@ def args_validation(args, argparser):
...
+            args.dry_run_no_signer
+        except AttributeError:
+            logger.info("args: dry_run_no_signer argument does not exist.")
+        else:
+            args.dry_run = args.dry_run_no_signer
```
Comment (jdsika) [inline src/launch_common.py:155]: is the "or args.dry_run" missing here?

**Comment by jdsika** [comment]
Please add your tz address and GitHub username in the contributors.csv

**Comment by jdsika** [comment]
This is an interesting situation. We add a new feature and I start to wonder on compatibility issues and how the combinations. It would be great if you could spend some effort on defining tests for these parameters that would give me confidence in the functionality and also help me to understand the impact on TRD

**Comment by nicolasochem** [comment]
Nice PR!

**Comment by jdsika** [comment]
replaced by #631

**Pull request closed without merging**